lib/souffle_lib.dl

```diff
@@ -2,26 +2,27 @@
 
 import intern
 import regex
+import souffle_types
 
-function castTo32(i: bit<64>): bit<32> = i[31:0]
-function castTo64(i: bit<32>): bit<64> = 32'd0 ++ i
+function castTo32(i: bit<64>): Tnumber = i[31:0] as Tnumber
+function castTo64(i: Tnumber): bit<64> = 32'd0 ++ (i as bit<32>)
```

Comment:
Likewise, this can now be written as `i as bit<64>`

Reply:
P4 forbids casts that change both the width and the sign, because the semantics is unclear.
I expect we will want to do the same thing.

Reply:
So, is it `i as bit<32> as bit<64>`?

Reply:
Should give the same result really.